### PR TITLE
GH-1670 Indented Tree

### PR DESCRIPTION
**Pull request by GordonSmith**
Added indented tree with nested XML support

Signed-off-by: Gordon Smith <gordonjsmith@gmail.com>

### CODE REVIEW

**Comment by GordonSmith** [comment]
@jchambers-ln Please Review:  http://rawgit.com/GordonSmith/Visualization/GH-1670-XMLTree/demos/dermatology.html?src/tree/Indented.xml

**Comment by jchambers-ln** [comment]
Looks pretty good, I think they may not like not being able to copy out text but we will see.

File: src/tree/Indented.js [hunk 1]
```diff
@@ -0,0 +1,237 @@
...
+                }
+            }
+        } else if (xml.nodeType === 3) { // text
+            retVal.label = xml.nodeValue;
+        }
+
+        if (xml.hasChildNodes()) {
+            for (var i = 0; i < xml.childNodes.length; i++) {
+                var item = xml.childNodes.item(i);
+                var child = xmlToJson(item, id + "[" + retVal.children.length + "]");
```
Comment (hamiltwp) [inline src/tree/Indented.js]: Is OK to have non-unique ID's in the JSON result?

Comment (GordonSmith) [inline src/tree/Indented.js]: yes - but not ideal, the ID is primarily to help with the animation (so the DOM nodes are re-used correctly and consistently)

**Pull request merged**